src/NuGet.Core/NuGet.Packaging/PackageFileExtractor.cs

```diff
@@ -35,7 +36,7 @@ private static HashSet<string> GatherIntellisenseXmlFiles(IEnumerable<string> pa
 
             foreach (var file in refAndLibFiles)
             {
-                if (!file.EndsWith(".xml"))
+                if (!file.EndsWith(".xml", StringComparison.Ordinal))
```

Comment:
This looks like needs to be file-system aware. But, PackageFileExtractor works on nupkgs, which has standarized filepaths.

I will look at this later to find out if we need FS-awareness or not.

```suggestion
                if (!file.EndsWith(".xml", PathUtil.GetStringComparisonBasedOnOS(()))
```